tests/Components/GenericExpressCheckout/CustomerRegistrationUtilTest.php

```diff
@@ -60,24 +61,28 @@ public function testIfAllRequiredKeysArePresentAndHaveValidValues(): void
         static::assertArrayHasKey('phone', $data['billingAddress']);
     }
 
-    public function testWithOnlyShippingAddress(): void
+    /**
+     * @dataProvider billingAddressDataProvider
+     * @testdox It returns correct data when shipping data is missing in response and billing data is available with prefix "$billingPrefix"
+     */
+    public function testItReturnsCorrectDataWhenShippingDataIsMissingInResponse(string $billingPrefix): void
     {
         /** @var CustomerRegistrationUtil $util */
         $util = $this->getContainer()->get(CustomerRegistrationUtil::class);
 
         $data = $util->getCustomerDataBagFromGetCheckoutSessionResponse([
             'addpaydata' => [
                 'email' => 'test@localhost.local',
-                'telephonenumber' => '0123456789',
```

Comment:
are these changes a mistake?
the method has tested shipping-data, and now it testing billing-data, but the method still is named to test shipping-data. 
i am confused :P

Reply:
I had fundamentally renamed the tests and adapted them to the new logic. The diff may be a little confusing here, but it should be correct. This function tests whether all data is correct if no shipping data is passed.